scalafix/v2/rules/src/main/scala/fix/v2.scala

```diff
@@ -0,0 +1,370 @@
+                            ),
+                            List(bec)
+                        ),
+                      Term.Name("flatMap")| Term.Name("map")
+                  ),
+            _) if isBlockingExecutionContext(bec) =>
+        resourceFMap += sr.pos
+        resourceSet += ap.pos
+        val tpe = firstType(getType(bec.symbol)).get
+
+          List(Patch.replaceTree(tpe2, "Blocker"),
+          Patch.replaceTree(bec, s"Blocker[$tpe]")).asPatch + addCatsEffectImports
+
+
+      case fy: Term.ForYield if hasStreamResource(fy.enums) =>
+        replaceBlockingExecutionContextForYield(fy.enums, resourceSet, resourceFMap)
+
+      case fy: Term.ForYield if hasStreamResourceWithType(fy.enums) =>
+        replaceBlockingExecutionContextWithTypeForYield(fy.enums, resourceSet, resourceFMap)
+
+      case resourceMatcher(ap @ Term.Apply(_, List(bec))) if isBlockingExecutionContext(bec) =>
+        if (resourceSet.contains(ap.pos))
+          Patch.empty
+        else {
+            val tpe =
+              bec match {
+                case Term.ApplyType(_, List(x)) => x
+                case _ => firstType(getType(bec.symbol)).get
+              }
+            Patch.replaceTree(bec, s"Blocker[$tpe]") + addCatsEffectImports
+         }
+      case linesAsyncMatcher(ap @ Term.Apply(_, List(_, bec))) =>
+        replaceBlockingExecutionContextTerm(bec, ap, resourceFMap)
+
+      case showLinesAsyncMatcher(ap @ Term.Apply(_, List(_, bec))) =>
+        replaceBlockingExecutionContextTerm(bec, ap, resourceFMap)
+
+      case showLinesStdOutAsyncMatcher(ap @ Term.Apply(_, List(bec))) =>
+        replaceBlockingExecutionContextTerm(bec, ap, resourceFMap)
+
+      case defaultMatcher(ap: Term.ApplyType) =>
+        addBlockerTerm(ap)
+
+      case readAllMatcher(ap @ Term.Apply(_,  List(_, bec, _))) =>
+        replaceBlockingExecutionContextTerm(bec, ap, resourceFMap)
+
+      case writeAllMatcher(ap @ Term.Apply(_,  List(_, bec))) =>
+        replaceBlockingExecutionContextTerm(bec, ap, resourceFMap)
+
+      case watcherMatcher(ap:Term.ApplyType) =>
+        addBlockerTerm(ap)
+
+      case watchMatcher(ap @ Term.Apply(fun, params)) =>
+        val newParams =
+          Term.Apply(
+            Term.Select(Term.Name("Blocker"), Term.Name("liftExecutionContext")),
+            List(
+              Term.Apply(
+                Term.Select(Term.Name("ExecutionContext"), Term.Name("fromExecutorService")),
+                  List(
+                    Term.Apply(
+                      Term.Select(Term.Name("Executors"), Term.Name("newCachedThreadPool")),
+                      List()
+                    )
+                  )
+              )
+            )
+          ):: params
+        Patch.replaceTree(ap, Term.Apply(fun, newParams).toString)
+
+      case fromPathMatcher(ap @ Term.Apply(fun, List(_, bec, _))) =>
+        replaceBlockingExecutionContextTermAndRenameFunction(bec, ap, resourceFMap)(fun, s"io.file.FileHandle.fromPath")
```

Comment:
This transformation will change the type, indeed in [this commit](https://github.com/functional-streams-for-scala/fs2/commit/bdcbb5b1c69d80a62d84c531bf7519aee13ac7f0#diff-88c75a27cf772b8d90dd50ba68691626L108):
```scala
io.file.pulls.fromPath(...): Pull[F, Nothing, Pull.Cancellable[F, FileHandle[F]]]
``` 
to
```scala
io.file.FileHandle.fromPath(...): Resource[F, FileHandle[F]]
```
But the user can use the compiler error to sort that further... Unless someone know a way to convert `Pull[F, Nothing, Pull.Cancellable[F, FileHandle[F]]]` into a `Resource[F, FileHandle[F]]` in which case the rule can be improved